sgt-back-end/index.js

```diff
@@ -0,0 +1,159 @@
+const pg = require('pg');
+const express = require('express');
+const app = express();
+
+app.use(express.json());
+
+// only create ONE pool for your whole server
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades', (req, res) => {
+
+  const sql = `
+    select "gradeId",
+           "name",
+           "course",
+           "score",
+           "createdAt"
+      from "grades"
+      `;
+  db.query(sql)
+    .then(result => {
+      const grade = result;
+      if (!grade) {
+        res.status(404).json({
+          Error: 'Cannot find grade'
+        });
+      } else {
+        res.json(grade.rows);
+      }
+
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occured'
+      });
+    });
+
+});
+
+app.post('/api/grades', (req, res) => {
+  const insName = req.body.name;
+  const insCourse = req.body.course;
+  const insScore = req.body.score;
+  if (!insName) {
+    return res.status(400).json({ error: 'Missing input for Name' });
+  } else if (insScore > 100 || insScore < 0) {
+    return res.status(400).json({ error: 'Invalid input for Score' });
+  } else if (!insCourse) {
+    return res.status(400).json({ error: 'Invalid input for Course' });
+  }
+
+  const sql = `
+    insert into "grades" ("name", "course", "score")
+    values ('${insName}', '${insCourse}', '${insScore}')
+    returning*;
+      `;
+  db.query(sql)
+    .then(result => {
+      const grade = result;
+      if (!grade) {
```

Comment:
When checking for a grade you want to check if a value was sent back, it will be the first value in the `rows` array. So to set `grade` you should do:
```JAVASCRIPT
const grade = result.rows[0];
```